src/aui-diagram-builder/tests/unit/js/aui-diagram-builder-overlapping-connectors-tests.js

```diff
@@ -0,0 +1,223 @@
+        var yOverlap = Math.max(0, Math.min(rectangle1.yMax, rectangle2.yMax) - Math.max(rectangle1.y, rectangle2.y));
+
+        if (xOverlap > 0 && yOverlap > 0) {
+            return true;
+        } else {
+            return false;
+        }
+    }
+
+    suite.add(new Y.Test.Case({
+        name: 'DiagramBuilderConnector',
+
+        'check for any interconnected connectors': function() {
+
+            var diagramBuilder = new Y.DiagramBuilder({
+                availableFields: [],
+                boundingBox: '#diagramBuilder',
+                fields: [
+                    {
+                        name: 'Start',
+                        type: 'condition',
+                        xy: [100, 100]
+                    },
+                    {
+                        name: 'End',
+                        type: 'condition',
+                        xy: [250, 100]
+                    }
+                ]
+            }).render();
+
+            diagramBuilder.connectAll([
+                {
+                    connector: { name: 'link test 1' },
+                    source: 'Start',
+                    target: 'End'
+                },
+                {
+                    connector: { name: 'link test 2' },
+                    source: 'End',
+                    target: 'Start'
+                }
+            ]);
+
+            var hasOverlapping = false;
+
+            diagramBuilder.eachConnector(function(connector) {
+                if (connector.hasConnectorsInterconnected()) {
+                    hasOverlapping = true;
+                }
+            });
+
+            Y.Assert.areEqual(hasOverlapping, true);
+        },
+
+        'check if the connectors are not visually overlapping the quadrant 0 and 4': function() {
+
+            var diagramBuilder3 = new Y.DiagramBuilder({
+                availableFields: [],
+                boundingBox: '#diagramBuilder',
+                fields: [
+                    {
+                        name: 'Start2',
+                        type: 'condition',
+                        xy: [100, 100]
+                    },
+                    {
+                        name: 'End2',
+                        type: 'condition',
+                        xy: [300, 100]
+                    }
+                ]
+            }).render();
+
+            diagramBuilder3.connectAll([
+                {
+                    connector: { name: 'Link test 1' },
+                    source: 'Start2',
+                    target: 'End2'
+                },
+                {
+                    connector: { name: 'Link1 test 2' },
+                    source: 'End2',
+                    target: 'Start2'
+                }
+            ]);
+
+            var hasOverlapping = false;
+            var connectors = [];
+
+            diagramBuilder3.eachConnector(function(connector) {
+                var _connector = connector.get('nodeName')._node;
```

Comment:
Hey @interaminense, you probably don't need to get _node. You can just pass the Alloy Node to rectOverlap. The Alloy Node has a method that can be useful to you: node.get('region'). It will return all the information you need to check if they overlap. It even has a method called inRegion([region]) check the YUI documentation for more details.

Reply:
Hey @brunobasto,
Thank you! I will try to verify the documentation of the YUI to use the methods correctly.